File: hrmsfrontend/src/services/socialMediaService.js
import axios from "axios";

export default class SocialMediaService {
  constructor() {
    this.apiUrl = "http://localhost:8080/api/socialMediasController/";
  }

  getAll() {
    return axios.get(`${this.apiUrl}getAll`);
  }

  getByJobSeekerId(jobSeekerId) {
    return axios.get(
      `${this.apiUrl}getByJobSeeker_id?jobSeekerId=${jobSeekerId}`
    );
  }

  add(socialMedia) {
    return axios.post(`${this.apiUrl}add`, socialMedia);
  }

  update(socialMedia) {
    return axios.post(`${this.apiUrl}add`, socialMedia);
  }
}

File: hrms/src/main/java/com/kaya/hrms/business/abstracts/SocialMediaService.java
package com.kaya.hrms.business.abstracts;

import java.util.List;

import com.kaya.hrms.core.utilities.results.DataResult;
import com.kaya.hrms.core.utilities.results.Result;
import com.kaya.hrms.entities.Dtos.SocialMediaDtos.SocialMediaAddDto;
import com.kaya.hrms.entities.Dtos.SocialMediaDtos.SocialMediaUpdateDto;
import com.kaya.hrms.entities.concretes.SocialMedia;

public interface SocialMediaService {
	
	DataResult<List<SocialMedia>> getAll();
	
	DataResult<List<SocialMedia>> getByJobSeeker_id(int jobSeekerId);
	
	Result add(SocialMedia socialMedia);

	Result update(SocialMedia socialMedia);
}
